cmd/dogstatsd/main.go

```diff
@@ -187,7 +187,7 @@ func start(cmd *cobra.Command, args []string) error {
 	}
 
 	aggregatorInstance := aggregator.InitAggregator(s, hname, "agent")
-	statsd, err := dogstatsd.NewServer(aggregatorInstance.GetChannels())
+	statsd, err := dogstatsd.NewServer(aggregatorInstance.GetBufferedChannels())
```

Comment:
It won't be a real problem but now that we're using buffered channels, we should probably take care of flushing those while closing the Dogstatsd server/aggregator in order to not lose any messages. It may have miss it but I didn't find a code path where it is done.

Reply:
We were already using buffered channel. Now we use buffered channels that send buffers. I agree `GetBufferedChannels` is a bit confusing... Can't think of a better name, does any come to your mind ?
I think there is a card on our sprint or backlog to decide what to do to avoid losing data on an agent exit.